### ISSUE TITLE
Common: Issues with strptime in wMessageStats.c

### ISSUE THREAD

**Issue by MattMulhern** [OpenMAMA/OpenMAMA#11]
**Replacing [bz-113](http://bugs.openmama.org/show_bug.cgi?id=113)**

There appears to be an issue with building wMessageStats.c on Windows, which generates the following warning:

```bash
wMessageStats.c(618): warning C4013: 'strptime' undefined; assuming extern returning int
```
Windows doesn't actually have a native strptime implementation, so common actually includes one in the windows compatibility code (windows/strptime.c). For some reason this has been commented out of the wMessageStats.c code, and thus we hit this warning. I believe this will likely still operate as expected on 32 bit machines as long as int's and pointers are the same size, and the strptime code is actually compiled into the common libraries (which should be the case in both SCons and VS builds). It'll simply resolve the function during link time later.

It is less likely to operate correctly on a 64 bit machine, where int's and pointers are likely to have different sizes. If strptime returns a larger pointer value, it may get truncated, and everything will die horribly.

The fix seems fairly obvious, however more testing/investigation is needed to understand why the code was changed in the first place.

**Comment by fquinner** [comment]
Just wired up this implementation and whipped up a few unit tests around it and it looks grand to me. Will likely be pushing in a fix tomorrow evening.

**Comment by fquinner** [comment]
Implemented in #207
